pkg/retry/retry.go

```diff
@@ -32,20 +32,43 @@ const (
 	DefaultMaxElapsedTime      = 1 * time.Minute
 )
 
-// Retry retries the given function DefaultRetryCount times while sleeping some time between unsuccessful attempts
-// if retryOnError returns true, continue retry, otherwise, return error
-func Do(operation func() error, retryOnError func(err error) bool) error {
-	defer trace.End(trace.Begin(""))
+// simplified config for configuring a back off object. Callers should populate and supply this to DoWithConfig
+type BackoffConfig struct {
+	InitialInterval     time.Duration
+	RandomizationFactor float64
+	Multiplier          float64
+	MaxInterval         time.Duration
 
-	var err error
-	var next time.Duration
+	// this field will indicate the maximum amount of "sleep" time that will occur.
+	MaxElapsedTime time.Duration
+}
 
-	b := &backoff.ExponentialBackOff{
+// Do retries the given function until DefaultMaxInterval time passes, while sleeping some time between unsuccessful attempts
+// if retryOnError returns true, continue retry, otherwise, return error
+func Do(operation func() error, retryOnError func(err error) bool) error {
+	bConf := &BackoffConfig{
 		InitialInterval:     DefaultInitialInterval,
 		RandomizationFactor: DefaultRandomizationFactor,
 		Multiplier:          DefaultMultiplier,
 		MaxInterval:         DefaultMaxInterval,
 		MaxElapsedTime:      DefaultMaxElapsedTime,
+	}
+
+	return DoWithConfig(operation, retryOnError, bConf)
+}
+
+// DoWithConfig will attempt an operation while retrying using an exponential back off based on the config supplied by the caller. The retry decider is the supplied function retryOnError
+func DoWithConfig(operation func() error, retryOnError func(err error) bool, config *BackoffConfig) error {
```

Comment:
talked to @caglar10ur and we have decided to make a `NewConfig` function which will populate the config object with all the defaults and let the user override them at will.